sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveDefaultColumns.scala

```diff
@@ -589,8 +631,10 @@ case class ResolveDefaultColumns(
     resolved.collectFirst {
       case r: UnresolvedCatalogRelation =>
         r.tableMeta.schema
-      case d: DataSourceV2Relation if !d.skipSchemaResolution && !d.isStreaming =>
-        CatalogV2Util.v2ColumnsToStructType(d.table.columns())
+      case DataSourceV2Relation(table: SupportsCustomSchemaWrite, _, _, _, _) =>
```

Comment:
We can create follow-up jira to cover the resolution of insertion with SupportsCustomSchemaWrite.

Reply:
👍  I created it here: https://issues.apache.org/jira/browse/SPARK-43358